nemo_rl/environments/genrm_environment.py

```diff
@@ -0,0 +1,277 @@
+            
+        elif num_responses == 2:
+            # Expected format: "[The Begin of Individual Scores]\n\\boxed{x, y} \n[The End of Individual Scores]\n[The Begin of Ranking Score]\n\\boxed{z} \n[The End of Ranking Score]"
+            
+            # Check individual scores section
+            individual_pattern = r'\[The Begin of Individual Scores\]\s*\n\s*\\boxed\{([^}]+)\}\s*\n\s*\[The End of Individual Scores\]'
+            individual_match = re.search(individual_pattern, assistant_response, re.DOTALL)
+            
+            if not individual_match:
+                return False, None, None, "Format violation: Missing or incorrect '[The Begin of Individual Scores]\\n\\\\boxed{x, y} \\n[The End of Individual Scores]' section"
+            
+            # Check ranking score section
+            ranking_pattern = r'\[The Begin of Ranking Score\]\s*\n\s*\\boxed\{([^}]+)\}\s*\n\s*\[The End of Ranking Score\]'
+            ranking_match = re.search(ranking_pattern, assistant_response, re.DOTALL)
+            
+            if not ranking_match:
+                return False, None, None, "Format violation: Missing or incorrect '[The Begin of Ranking Score]\\n\\\\boxed{z} \\n[The End of Ranking Score]' section"
+            
+            # Extract individual scores
+            scores_content = individual_match.group(1).strip()
+            individual_scores = [score.strip() for score in scores_content.split(',')]
+            
+            # For two responses, should have exactly two scores
+            if len(individual_scores) != 2:
+                return False, None, None, f"Format violation: Expected exactly 2 individual scores, found {len(individual_scores)}"
+            
+            # Extract preference ranking
+            preference_content = ranking_match.group(1).strip()
+            # Preference ranking should be a single value
+            if ',' in preference_content:
+                return False, None, None, "Format violation: Preference ranking should be a single value, found comma"
+            
+            return True, individual_scores, preference_content, ""
+        
+        else:
+            return False, None, None, f"Unsupported number of responses: {num_responses}"
+    
+    def step(self, message_log_batch: list[list[dict[str, str]]], metadata: list[GenRMEnvironmentMetadata],) -> EnvironmentReturn:
+        """Evaluate GenRM predictions and return rewards."""
+        
+        rewards = []
+        observations = []
+        
+        for conversation, meta in zip(message_log_batch, metadata):
+            # Extract assistant's response
+            assistant_response = ""
+            for msg in conversation:
+                if msg["role"] == "assistant":
+                    assistant_response = msg["content"]
+                    break
+            
+            
+            
+            is_valid_format, individual_scores, preference_ranking, error_message = self.validate_format_and_extract(
```

Comment:
Btw it looks like the code isn't cleanly formatted => I suggest to setup the pre-commit hooks as described on https://github.com/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md which I expect should run auto-formatting on commit (I haven't tried it on NeMo-RL but that's how it should work).